lib/taskana-core-test/src/test/java/acceptance/classification/query/ClassificationQueryPaginationAccTest.java

```diff
@@ -0,0 +1,139 @@
+import org.junit.jupiter.api.Test;
+
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.models.ClassificationSummary;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.security.WithAccessId;
+
+@TaskanaIntegrationTest
+public class ClassificationQueryPaginationAccTest {
+
+  @TaskanaInject ClassificationService classificationService;
+
+  @WithAccessId(user = "businessadmin")
+  @BeforeAll
+  void createClassifications() throws Exception {
+    for (int i = 0; i <= 18; i++) {
+      defaultTestClassification().buildAndStoreAsSummary(classificationService);
+    }
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetFirstFiveClassifications_When_OffsetIsZeroAndLimitIsFive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(0, 5);
+    assertThat(results).hasSize(5);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetLastFourClassifications_When_OffsetIsFifteenAndLimitIsFive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(15, 5);
+    assertThat(results).hasSize(4);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetZeroClassifications_When_OffsetIsNegativeAndLimitIsNegative() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(-1, -1);
+    assertThat(results).hasSize(0);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetZeroClassifications_When_LimitIsNegative() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(0, -1);
+    assertThat(results).hasSize(0);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetZeroClassifications_When_OffsetAndLimitAreZero() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(0, 0);
+    assertThat(results).hasSize(0);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetOneClassifications_When_OffsetIsNegativeAndLimitIsPositive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").list(-1, 1);
+    assertThat(results).hasSize(1);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetFirstFiveClassificationsOnAPage_When_FirstPageAndPageSizeIsFive() {
+    List<ClassificationSummary> results =
+        classificationService.createClassificationQuery().domainIn("DOMAIN_A").listPage(1, 5);
+    assertThat(results).hasSize(5);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_GetLastFourClassificationsOnAPage_When_FourthPageAndPageSizeIsFive() {
```

Comment:
What do you mean with "PageSize" in the test name? Is it the same as "Limit" in previous test names? If I understand this test correctly, the size of the last page is four. That's why the test name is confusing to me

Reply:
The size of last page is four because there are not enough tasks